### DISCUSSION TITLE
Installing on a NVIDIA Jetson Nano

### DISCUSSION THREAD

**Discussion by tomsepe** [nerdyrodent/VQGAN-CLIP#101] [Q&A]
I've been having some issues trying to get this to work on the Jetson Nano. It seems like it would be a great candidate for this since it has 128 GPU cores.

Firstly I had to tweak some of the versions in the requirements.txt as some of those versions don't seem to be available or compatible on the Nano. For instance I had to downgrade to numpy1.19.4 cause that works on the Nano.

So I finally got he script to run...

But the big issue I seem to be running into now is that the software doesn't see the GPU.

I got it to see the GPU once. I think with torch 1.8.0 and CUDA 11.0 or 10.2 but it failed because I think the script needs torch 1.9.0 to function.  But everytime I install torch 1.9.0 I loose CUDA support.

torch.cuda.is_available() keeps coming up false.

maybe this just won't work on the Jetson Nano?

I've tried running it with just the CPU, but  it gets as far as:
`Restored from checkpoints/vqgan_imagenet_f16_16384.ckpt`
waits for a little while then I get:
`Illegal instruction (core dumped)`

and I'm not advanced enough to be able to troubleshoot that.

any thoughts / insights / ideas appreciated.

**Comment by znmeb** [comment]
You can install PyTorch 1.9.0 from NVIDIA-built wheels - see https://forums.developer.nvidia.com/t/pytorch-for-jetson-version-1-10-now-available/72048/500

**Reply by tomsepe** [reply]
Thank you. I did install from the pre-built wheel, but when I do install 1.9.0 then I loose CUDA support and the GPU doesn't show up! 

I'll probably ditch this direction, since there just seems to be a bunch of issues getting it to work and ultimately. the Jeston Nano is a pretty limited computer, although once could upgrade to the Jetson AGX Xavier...but they are $2k and 34 week lead time...   I have access to some windows machines that have some phat graphics cards and large RAM. So that's a much more accesible and affordable option for now!

**Reply by znmeb** [reply]
> although once could upgrade to the Jetson AGX Xavier...but they are $2k and 34 week lead time... 

That's because legitimate sellers are out of stock on the AGX Xavier and Xavier NX - the people selling them are scalpers. I saw them on Amazon; Amazon should know better!
 
>I have access to some windows machines that have some phat graphics cards and large RAM. So that's a much more accesible and affordable option for now!

Here's the way to get a GPU-enabled PyTorch on an x86_64 machine: https://pytorch.org/get-started/locally/
